buildSrc/src/main/java/quilt/internal/tasks/lint/FindDuplicateMappingFilesTask.java

```diff
@@ -0,0 +1,161 @@
+package quilt.internal.tasks.lint;
+
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.Multimap;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.GradleException;
+import org.gradle.api.file.DirectoryProperty;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.gradle.api.tasks.InputDirectory;
+import org.gradle.api.tasks.TaskAction;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.util.*;
+import java.util.regex.Pattern;
+import java.nio.file.Path;
+import java.util.stream.Stream;
+
+public abstract class FindDuplicateMappingFilesTask extends DefaultTask {
```

Comment:
Could this be a `Checker` and not a full task?

Reply:
Sure, just... let me figure out what a `Checker` is

Reply:
Its part of the buildSrc, should be a bit cleaner.

Reply:
It looks like `Checker`s only see one mapping at a time, so I'd have to maintain `allMappings` and `duplicateMappings` in static fields, or add methods to `Checker` to allow attaching things to a task run.
It also doesn't look like they expose the actual mapping file, so I couldn't do the extension check without adding methods to `Checker` for it.
Is that preferable to a separate task?

Reply:
I'll start implementing it and see how it looks

Reply:
conclusion: it can't be a `Checker` because enigma doesn't create multiple entries if multiple files map a class